### DISCUSSION TITLE
[docker-compose] Is there jdbcDriver in oap image?

### DISCUSSION THREAD

**Discussion by yixiubest** [apache/skywalking#9893] [Q&A]
If I start oap by using mysql-storage, I need to add mysql-connector-java-5.1.47.jar into folder oap-libs. 
Now question is if I start oap by docker-compose and use mysql-storage also, how can I add mysql-connector dependency into image? Or do I need to do this?

**Comment by kezhenxu94** [comment]
You need to add mysql driver if you want to use mysql as storage, about how to add the mysql driver to the Docker container, refer to https://github.com/apache/skywalking/blob/828e6e2f2b57a0f06bb0d507e3296d2377943d9a/test/e2e-v2/cases/storage/mysql/docker-compose.yml#L44

**Reply by yixiubest** [reply]
When I tried to follow the doc and started the image, I got an error "sh: 1: /download-mysql.sh: not found". So where is the file?
![image](https://user-images.githubusercontent.com/38375081/199376190-ea2e2ca8-c9f9-440b-bb4c-fc1d34d5b707.png)

**Reply by yixiubest** [reply]
@kezhenxu94 Or where do I need to put it?

**Reply by kezhenxu94** [reply]
Please, you need to download `download-mysql.sh` and mount it into the container, learn the full example I posted above, don't simply copy one line and expect it works

**Reply by yixiubest** [reply]
The example needs network support, but I need to deploy in an offline environment, Is there any case? @kezhenxu94

**Reply by kezhenxu94** [reply]
> The example needs network support, but I need to deploy in an offline environment, Is there any case? @kezhenxu94

You can download the mysql-connector.jar and mount it into the oap folder inside the container

**Comment by wu-sheng** [comment]
The docker hub page provides a very clear way. https://hub.docker.com/r/apache/skywalking-oap-server Try to read docs and understand things, before simply asking. This is not a complex issue, just a very basic docker practice.

## Extend image
If you intend to override or add config files in /skywalking/config, /skywalking/ext-config is the location for you to put extra files. The files with the same name will be overridden, otherwise, they will be added in /skywalking/config.

If you want to add more libs/jars into the classpath of OAP, for example, new metrics for OAL. These jars can be mounted into /skywalking/ext-libs, then entrypoint bash will append them into the classpath. Notice, you can't override an existing jar in classpath.
